test/Tests/TestPartyCreationMenu.cpp

```diff
@@ -411,11 +449,29 @@ GAME_TEST(Issue, Issue503) {
     check503health({ {0, 1147}, {1, 699}, {2, 350}, {3, 242} });
 }
 
+GAME_TEST(Issue, Issue504) {
+    // Going to prison doesn't recharge hirelings
+    test->playTraceFromTestData("issue_504.mm7", "issue_504.json");
+    EXPECT_TRUE(pParty->pPartyBuffs[PARTY_BUFF_HEROISM].Active());
+}
+
 GAME_TEST(Issue, Issue506) {
     // Check that scroll use does not assert
     test->playTraceFromTestData("issue_506.mm7", "issue_506.json");
 }
 
+GAME_TEST(Issues, Issue520) {
+    // Party should take fall damage
+    uint64_t oldHealth = 0;
+    test->playTraceFromTestData("issue_520.mm7", "issue_520.json", [&] { oldHealth = GetPartyHealth(); });
+    EXPECT_LT(GetPartyHealth(), oldHealth);
+}
+
+GAME_TEST(Issues, Issue521) {
+    // 500 endurance leads to asserts in Player::SetRecoveryTime
+    test->playTraceFromTestData("issue_521.mm7", "issue_521.json");
```

Comment:
Maybe we should check that 500 endurance lets the chars skip recovery altogether?
E.g. check for active char before and after receiving damage?